### ISSUE TITLE
systemd taking a lot of time during boot if it doesn't get network access

### ISSUE THREAD

**Issue by plomari** [systemd/systemd#27257]
### systemd version the issue has been seen with

252.6

### Used distribution

Debian unstable

### Linux kernel version used

5.10.176

### CPU architectures issue was seen on

x86_64

### Component

systemd

### Expected behaviour you didn't see

If there's a boot problem, systemd should give me the choice to continue booting and skipping whatever init thing causes problem, or dropping into a root shell.

### Unexpected behaviour you saw

systemd waits several minutes (maybe just 1m30s, I don't remember) waiting for something that won't happen, then apparently gives up and continues booting. There doesn't seem to be any way to interact with.

### Steps to reproduce the problem

Apparently systemd wants network access during boot, maybe to sync the clock with NTP. It appears that sometimes, the "stable" network interface names are not stable and change on boot. (Either it's random, or it happened with a minor upgrade within the 5.10 stable kernel series. All my network interfaces are connected with PCIe and no hotplug is going on, the hardware wasn't changed between the reboots.)

Due to the mismatching names, my systemd-networkd configuration broke. systemd couldn't access the network. I suspect the timesyncd service stalled or failed.

You may be able to reproduce the problem by taking your network card and plugging it into a different PCIe slot (or doing the equivalent with a VM). Or maybe you can just try to boot with the network cable unplugged. (I wonder how this works on laptops, they usually will not have network access during boot...)

This specific problem is caused by networkd or timesyncd. But fundamentally there's a worse problem with systemd making you wait like an idiot until it's done with its timeout, with no way to intervene. There might be worse situations, which leave you with a hard to recover system.

**Comment by YHNdnzj** [comment]
systemd-timesyncd doesn't wait for network, unless you've enabled `systemd-time-wait-sync.service`. That is, systemd doesn't block your boot process, but your dependencies do.

**Comment by poettering** [comment]
This seems like a local configuration issue. systemd-network-wait-online is normally not part of the boot process, except if some local component pulls it explicitly in, for example because you have an unconditional NFS mount or so in /etc/fstab, or something like iscsi configured. Because if you have that we *have* to wait for the network to be configured before we can continue booting.

systemd-timesync codes not wait for the network. 

systemd is not an interactive tool, for security reasons: we cannot allow unauthenticated users to make changes to the system (except maybe ask for a reboot or so). Hence we do not provide an interactive UI you cold use to interfere with the boot process. 

If you like you can bind a debug shell to kbrequest.target (i.e. ctlr-alt-up), but this is not something we can generally do as it opens a major security hole. This would give you a way to start a debug shell any time on VT9.

Hence, I really don't think there's anything to fix here on systemd's side: something asked for the boot to be blocked until the network is up. turn that off and things should be fine. systemd does not do that by default, it does so only on request (i.e. via  dep) of some other component. But unconditionally providing an interactive boot is not really an option.

You can also do "systemctl disable systemd-network-wait-online" (and there's a similar thing for NM), which will make "network-online.target" a NOP. But if you do that then of course the services that asked for this will fail...

for details on figuring out what component is responsible for pulling in network-online.target see: https://freedesktop.org/wiki/Software/systemd/Debugging/#diagnosingbootproblems

But at that points this becomes more of a support question, hence please continue on the systemd ML, not here (here we only track bugs and feature reuqests, not support questions). hence closing.
